content/docs/examples/advanced-egress/egress-gateway-mtls-origination/index.md

```diff
@@ -0,0 +1,563 @@
+---
+title: Mutual TLS origination by Egress Gateway
+description: Describes how to configure an Egress Gateway to perform mutual TLS origination to external services.
+weight: 45
+keywords: [traffic-management,egress]
+---
+
+The [Configure an Egress Gateway](/docs/examples/advanced-egress/egress-gateway) example describes how to configure
+Istio to direct the egress traffic through a dedicated service called _Egress Gateway_. This examples shows how to
+configure an Egress Gateway to perform mutual TLS origination to external services.
+You will deploy an [NGINX](https://www.nginx.com/) server in your Kubernetes cluster without injecting Istio sidecar
+proxy into it. This server will simulate a host `nginx.example.com` outside the Istio service mesh.
+
+Then you will configure the Egress Gateway to perform mutual TLS with this server.
+Finally, you will direct the traffic from the application pods inside the mesh to the server outside the mesh through
+the Egress Gateway. The Egress Gateway will perform mutual TLS origination with the NGINX server.
+
+## Generate client and server certificates and keys
+
+Generate the certificates and keys in the same way as in the [Securing Gateways with HTTPS](/docs/tasks/traffic-management/secure-ingress/#generate-client-and-server-certificates-and-keys).
+
+1.  Clone the <https://github.com/nicholasjackson/mtls-go-example> repository:
+
+    {{< text bash >}}
+    $ git clone https://github.com/nicholasjackson/mtls-go-example
+    {{< /text >}}
+
+1.  Change directory to the cloned repository:
+
+    {{< text bash >}}
+    $ pushd mtls-go-example
+    {{< /text >}}
+
+1.  Generate the certificates for `nginx.example.com`.
+    Use any password with the following command:
+
+    {{< text bash >}}
+    $ ./generate.sh nginx.example.com <password>
+    {{< /text >}}
+
+    When prompted, select `y` for all the questions.
+
+1.  Move the certificates into `nginx.example.com` directory:
+
+    {{< text bash >}}
+    $ mkdir ~+1/nginx.example.com && mv 1_root 2_intermediate 3_application 4_client ~+1/nginx.example.com
+    {{< /text >}}
+
+1.  Change directory back:
+
+    {{< text bash >}}
+    $ popd
+    {{< /text >}}
+
+## Deploy an NGINX server
+
+1.  Create a namespace `mesh-external` to represent services outside the Istio mesh. Note that the sidecar proxy will
+    not be automatically injected into the pods in this namespace since the automatic sidecar injection was not
+    [enabled](/docs/setup/kubernetes/sidecar-injection/#deploying-an-app) on it.
```

Comment:
Regarding the passive voice - I used it on purpose here. The goal is to spare the reader unnecessary details about who actually injects the sidecar proxy.

See passive voice https://istio.io/docs/setup/kubernetes/sidecar-injection/:

> The controller resource is unmodified. Sidecars can be updated selectively ...

In this case:
> Istio doesn't automatically inject the sidecar proxy into the pods 

Note that it is not Istio that injects the proxy, it is a mechanism in Kubernetes, which is not relevant to the example. The point of the sentence is that the sidecar proxies will not be injected, the deployed pods will not have sidecar proxies and will not be a part of the Istio mesh.

The same regarding:

>  Ensure that the key and the certificate are successfully loaded

vs. 

> Ensure that Kubernetes loaded the key and the certificates...

The passive voice means it does not matter who and how loaded the key/certificate, the reader should just verify that the key/certificate is there, in the file system of the pod.

@geeknoid @frankbu Could you please provide your opinion about the passive voice usage in istio.io ? I propose to use the passive voice to spare the readers unnecessary
technical details.

Reply:
@vadimeisenbergibm I don't have a strong opinion on the passive voice question. I think that passive voice can have a tendency to not flow as smoothly, but I understand your wanting to not say things that may imply things that are inaccurate or irrelevant to the task at hand. That said, these tasks are not legal documents, so saying things like "Istio will do ..." is generally understood to mean something will happen because of Istio, not necessarily that Istio will literally do it.

Reply:
@frankbu @geeknoid @rcaballeromx Additional point - I think Istio is one of the most complex software systems existing. You have Kubernetes (which is complex in itself) and you have Istio on top of it, and you have interplay of the layers (Kubernetes/Istio), and you may have Istio on top of frameworks other than Kubernetes. The amount of information, multiple configuration items, interplay of Kubernetes and Istio can be overwhelming for the users. In my opinion, we should spare the users any irrelevant information, and also prevent any distraction and confusion.